bin/inspec

```diff
@@ -106,5 +122,42 @@ class InspecCLI < Thor
   def version
     puts Inspec::VERSION
   end
+
+  private
+
+  def diagnose
+    return unless opts['diagnose']
+    puts "InSpec version: #{Inspec::VERSION}"
+    puts "Train version: #{Train::VERSION}"
+    puts 'Command line configuration:'
+    pp options
+    puts 'JSON configuration file:'
+    pp options_json
+    puts 'Merged configuration:'
+    pp opts
+  end
+
+  def opts
+    # argv overrides json
+    Thor::CoreExt::HashWithIndifferentAccess.new(options_json.merge(options))
+  end
+
+  def options_json
+    conffile = options['json_config']
+    return read_config(conffile) if conffile
+
+    {}
+  end
+
+  def read_config(file)
+    if file == '-'
+      puts 'WARN: reading JSON config from standard input' if STDIN.tty?
```

Comment:
@arlimus when a user passes `--json-config=-`, she will just see that nothing happens. With this warning, the user will get an indication what is expected of her at the moment (i.e. provide JSON from the tty). It just affects the case of "reading from stdin when the session is interactive" (as opposed to non-interactive, i.e. `cat config.json | inspec detect --json-config=-`